ui/app/src/Views/AlertList/FilterBar.elm

```diff
@@ -1,20 +1,137 @@
+import Utils.Filter exposing (Matcher)
+import Views.AlertList.Types exposing (AlertListMsg(..))
+import Types exposing (Msg(..))
+import Json.Decode as Json
 
 
-view : String -> (String -> msg) -> msg -> Html msg
-view filterText inputChangedMsg buttonClickedMsg =
-    div [ class "input-group" ]
-        [ span [ class "input-group-addon" ]
-            [ i [ class "fa fa-filter" ] []
+viewMatcher : Matcher -> Html Msg
+viewMatcher matcher =
+    div [ class "col col-auto", style [ ( "padding", "5px" ) ] ]
+        [ div [ class "btn-group" ]
+            [ button
+                [ class "btn btn-outline-info"
+                , onClick (DeleteFilterMatcher True matcher |> MsgForAlertList)
+                ]
+                [ text <| Utils.Filter.stringifyMatcher matcher
+                ]
+            , button
+                [ class "btn btn-outline-danger"
+                , onClick (DeleteFilterMatcher False matcher |> MsgForAlertList)
+                ]
+                [ text "×" ]
             ]
-        , input
-            [ class "form-control", value filterText, onInput inputChangedMsg ]
-            []
-        , span
-            [ class "input-group-btn" ]
-            [ button [ class "btn btn-primary", onClick buttonClickedMsg ] [ text "Filter" ] ]
         ]
+
+
+lastElem : List a -> Maybe a
+lastElem =
+    List.foldl (Just >> always) Nothing
+
+
+viewMatchers : List Matcher -> List (Html Msg)
+viewMatchers matchers =
+    matchers
+        |> List.map viewMatcher
+
+
+onKey : String -> Int -> Msg -> Attribute Msg
+onKey event key msg =
+    on event
+        (Json.map
+            (\k ->
+                if k == key then
+                    msg
+                else
+                    Noop
+            )
+            keyCode
+        )
+
+
+view : List Matcher -> String -> Bool -> Html Msg
+view matchers matcherText backspacePressed =
+    let
+        className =
+            if matcherText == "" then
+                ""
+            else
+                case maybeMatcher of
+                    Just _ ->
+                        "has-success"
+
+                    Nothing ->
+                        "has-danger"
+
+        maybeMatcher =
+            Utils.Filter.parseMatcher matcherText
+
+        onKeydown =
+            onKey "keydown" 8 <|
+                case ( matcherText, backspacePressed ) of
+                    ( "", True ) ->
+                        Noop
+
+                    ( "", False ) ->
+                        lastElem matchers
+                            |> Maybe.map (DeleteFilterMatcher True >> MsgForAlertList)
+                            |> Maybe.withDefault Noop
+
+                    _ ->
+                        PressingBackspace True |> MsgForAlertList
+
+        onKeypress =
+            maybeMatcher
+                |> Maybe.map (AddFilterMatcher True >> MsgForAlertList)
+                |> Maybe.withDefault Noop
+                |> onKey "keypress" 13
```

Comment:
Could we move the magic numbers into constants? I'll forget every time that `13` is the enter keycode :)